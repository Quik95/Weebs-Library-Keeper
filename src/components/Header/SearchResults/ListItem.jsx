import React, { memo } from "react";
import PropTypes from "prop-types";

// MUI
import MuiListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import Avatar from "@material-ui/core/Avatar";
import ListItemText from "@material-ui/core/ListItemText";
import IconButton from "@material-ui/core/IconButton";

// temporary
import Icon from "@material-ui/icons/BeachAccess";

// Icons
import AddIcon from "@material-ui/icons/Add";

// styles
import useStyles from "./SearchResults.styles";

const ListItem = memo(function ListItem({ data }) {
  const classes = useStyles();
  const season = data.season || "??";
  const startYear = data.startDate.year || "??";
  const title = data.title.romaji || "??";

  return (
    <MuiListItem data-testid="ListItem">
      <ListItemAvatar>
        <Avatar
          alt={title}
          src={data.coverImage.medium}
          variant="square"
          className={classes.avatarSize}
        />
      </ListItemAvatar>
      <ListItemText
        primary={title}
        secondary={`${season} - ${startYear}`}
        className={classes.listItemText}
      />
      <IconButton>
        <AddIcon />
      </IconButton>
    </MuiListItem>
  );
});

ListItem.propTypes = {
  data: PropTypes.object.isRequired
};

export default ListItem;
