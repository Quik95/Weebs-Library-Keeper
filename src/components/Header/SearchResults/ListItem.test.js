import ListItem from './ListItem';

describe('<ListItem/>', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<ListItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
