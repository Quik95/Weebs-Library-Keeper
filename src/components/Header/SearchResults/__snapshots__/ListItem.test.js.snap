// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem/> Renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  data-testid="ListItem"
>
  <WithStyles(ForwardRef(ListItemAvatar))>
    <WithStyles(ForwardRef(Avatar))>
      <BeachAccessIcon />
    </WithStyles(ForwardRef(Avatar))>
  </WithStyles(ForwardRef(ListItemAvatar))>
  <WithStyles(ForwardRef(ListItemText))
    className="makeStyles-listItemText-2"
    primary="Dr. Stone"
    secondary="21/11/2019 18:50"
  />
  <WithStyles(ForwardRef(IconButton))>
    <AddIcon />
  </WithStyles(ForwardRef(IconButton))>
</WithStyles(ForwardRef(ListItem))>
`;
